### PR TITLE
fix: navigate to top of the page on route change

### LINKED ISSUE #1011: Navigating to another page does not place you at the top of that page
# Description
When navigating to another page you do not start at the top of the new page, see the video below for a visual explanation.

[Kooha-2022-09-25-12:22:38.webm](https://user-images.githubusercontent.com/31519867/192138932-c846b449-29af-4d94-8984-882d7fe05b5b.webm)

## To reproduce
- visit: http://localhost:3000/planters/940
- scroll down to the tree section and click a tree

---

Some hints:

- Please read our [readme](https://github.com/Greenstand/treetracker-web-map-client#treetracker-web) for more information/guide/tutorial.
- For design guidelines checkout our [Figma design](https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap).
- Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
- To know more about our organization, visit our [website](https://greenstand.org).
- If you want to join the slack community (some resources need the community member's permission), please leave your email address.
- More issues in this project: https://github.com/orgs/Greenstand/projects/26

### PR DESCRIPTION
**Pull request by Mohmn**
# Description

[comment]: # 'Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.'

Fixes #1011

## Type of change

[comment]: # 'Please delete options that are not relevant.'

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots

[comment]: # 'Please include screenshots of your changes if relevant.'

|       Before        |       After        |
| :-----------------: | :----------------: |
| "screenshot before" | "screenshot after" |

# How Has This Been Tested?

- [ ] Cypress integration
- [ ] Cypress component tests
- [ ] Jest unit tests

# Checklist:

- [x ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/LayoutMobile.js [hunk 1]
```diff
@@ -69,7 +70,9 @@ export default function Layout({ children }) {
         >
           <App />
         </Box>
-        <Drawer>{children}</Drawer>
+        <Drawer>
+          <Box ref={ref}>{children}</Box>
```
Comment (RubenSmn) [inline src/components/LayoutMobile.js:74]: Can we forward this ref to the `Drawer` so we can target the element that hold the `children`. Currently it does not scroll all the way to the top, there is still a "gap"

**Review by RubenSmn — CHANGES_REQUESTED** [review]

File: src/components/LayoutMobile.js (hunk 1 above)
Comment (Mohmn) [reply src/components/LayoutMobile.js]: done

**Comment by dadiorchen** [comment]
@RubenSmn but I remember you recommanded another way to solve this? a built-in support for this?

**Comment by RubenSmn** [comment]
@dadiorchen yes this PR uses the build-in functionality but we still needed some tweaks since our body/page does not contain the scrollable content. We have our drawer on mobile en the like side bar on desktop

File: src/components/Drawer.js [hunk 2]
```diff
@@ -301,7 +301,10 @@ export default function Drawer(props) {
         </Box>
 
         <Box
-          ref={contentRef}
+          ref={(el) => {
```
Comment (RubenSmn) [inline src/components/Drawer.js:304]: That is a clever way to solve this! I learned something!

**Review by RubenSmn — APPROVED** [review]
Awesome work!

**Comment by dadiorchen** [comment]
Great, give me some time to ponder on it

**Pull request merged**